micro_sam/prompt_generators.py

```diff
@@ -5,15 +5,24 @@
 class PointAndBoxPromptGenerator:
     def __init__(self, n_positive_points, n_negative_points, dilation_strength,
                  get_point_prompts=True, get_box_prompts=False):
-        self.n_positive_points = n_positive_points
-        self.n_negative_points = n_negative_points
         self.dilation_strength = dilation_strength
         self.get_box_prompts = get_box_prompts
         self.get_point_prompts = get_point_prompts
 
         if self.get_point_prompts is False and self.get_box_prompts is False:
             raise ValueError("You need to request for box/point prompts or both")
 
+        # pass an integer to allow deterministic prompting or list to randomise it
+        if isinstance(n_positive_points, int):
+            self.n_positive_points = n_positive_points
+        else:
+            self.n_positive_points = np.random.randint(n_positive_points[0], n_positive_points[1] + 1)
```

Comment:
You should do this in `__call__` and not in `__init__`!
For what we have in mind here we want to draw a random number of points from the range **for each object**.
With your current changes you set this to some fixed number for the whole generator class.